Remove BuiltInMeters()

martincostello opened the pull request:
Remove some extra `BuiltInMeters()` code that snuck in after #1948.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/2900)

martincostello committed:
Remove BuiltInMeters()
Remove some extra `BuiltInMeters()` code that snuck in after #1948.

eerhardt commented on playground/seq/Seq.ServiceDefaults/Extensions.cs:
```diff
@@ -46,8 +46,9 @@ public static IHostApplicationBuilder ConfigureOpenTelemetry(this IHostApplicati
             {
-                metrics.AddRuntimeInstrumentation()
-                       .AddBuiltInMeters();
+                metrics.AddAspNetCoreInstrumentation()
```
Any chance that while you are editing these files, can you sync them up to the latest template content? There are a couple other diffs.

Not saying do it for all the Extensions.cs files (but I won't stop you from doing it 😉) I just thought that since we are modifying these files anyway....

martincostello replied:
Sure - can you point me at the bit you mean just so I get the right thing and I'll do that shortly.

eerhardt replied:
Ah, sorry. I should have linked.

https://github.com/dotnet/aspire/tree/main/src/Aspire.ProjectTemplates/templates/aspire-servicedefaults

is the "source of truth" for what the ServiceDefaults template is.

martincostello replied:
I _think_ I've got them all - I just copied the edits from #2864.

martincostello committed:
Sync playgrounds with template
Apply review feedback.

eerhardt commented on playground/Playground.ServiceDefaults/Extensions.cs:
```diff
@@ -59,8 +64,11 @@ public static IHostApplicationBuilder ConfigureOpenTelemetry(this IHostApplicati
+                    //.AddGrpcClientInstrumentation()
+                    .AddHttpClientInstrumentation()
+                    // Add instrumentation for the AWS .NET SDK.
+                    .AddAWSInstrumentation();
```
Can't use AWS here

martincostello replied:
Sorry - copy-paste fail.

eerhardt commented on playground/orleans/OrleansServiceDefaults/Extensions.cs:
```diff
@@ -60,8 +65,9 @@ public static IHostApplicationBuilder ConfigureOpenTelemetry(this IHostApplicati
                 tracing.AddAspNetCoreInstrumentation()
-                       .AddGrpcClientInstrumentation()
-                       .AddHttpClientInstrumentation();
+                    // Uncomment the following line to enable gRPC instrumentation (requires the OpenTelemetry.Instrumentation.GrpcNetClient package)
+                    //.AddGrpcClientInstrumentation()
```
Since the TestShop does in fact use Grpc, can we keep this one? That way we get the right tracing in the TestShop.

eerhardt reviewed (commented):
Thanks for taking them all!

martincostello committed:
Fix-up extensions
- Add gRPC back to the test shop.
- Remove incorrect AWS references.

martincostello committed:
Restore Grpc reference

Put it back so it builds...

martincostello committed:
Restore package reference

Add another one back.

eerhardt committed:
PR feedback

- Remove OpenTelemetry.Instrumentation.Process from Directory.Packages.props
- Sync a few more ServiceDefaults with the template

eerhardt reviewed (approved):
LGTM. Thanks for the contribution!
